app/common/console_output.py

```diff
@@ -0,0 +1,102 @@
+from app.common.console_output_segment import ConsoleOutputSegment
+
+
+class ConsoleOutput(object):
+    """
+    Represents the console output of an atom.
+    """
+    def __init__(self, path):
+        """
+        :param path: sys path to the console output file
+        :type path: str
+        """
+        self.path = path
+
+    def segment(self, max_lines=50, offset_line=None):
+        """
+        Return a segment of the console output.
+
+        :param max_lines:
+        :type max_lines: int
+        :param offset_line: The starting line number in the console output from which the segment should
+            return content for. If set to None, then reads content starting from the end.
+        :type offset_line: int | None
+        :rtype: ConsoleOutputSegment
+        """
+        if offset_line is None:
+            return self._parse_from_end(max_lines)
+        else:
+            return self._parse_from_offset(max_lines, offset_line)
+
+    def _parse_from_offset(self, max_lines, offset_line):
+        """
+        Return up to max_lines of output starting from line number offset_line. If max_lines + offset_line
+        doesn't exist on this console output, then return everything beyond offset_line.
+
+        :type max_lines: int
+        :type offset_line: int
+        :rtype: ConsoleOutputSegment
+        """
+        total_lines = 0
+        output_lines = 0
+        console_output = []
+
+        with open(self.path, 'r') as f:
+            # Iterate up to the index offset_line
+            for i in range(0, offset_line):
+                # This is an error, meaning that there aren't even offset_line+1 lines in self.path.
+                if f.readline() == '':
+                    raise ValueError('offset: {} is higher than the total number of lines in file {}'.format(
+                        offset_line, self.path))
+
+                total_lines += 1
+
+            # Retrieve the console_output just between offset_line and offset_line + max_lines
+            for i in range(offset_line, offset_line + max_lines):
+                line = f.readline()
+
+                # We have reached the end of the file, or a line that has not finished being written to.
+                if line == '' or not line.endswith("\n"):
```

Comment:
I modified a PHPUnit test to perform a "sleep(60)" on the third test case, and did this:

phpunit test/unit/lib/Jira_Test.php > whatever.txt

and then cntrl-c'd out of it a few seconds into it.
```
taejun@mbp-002672 devtools (dev) $ cat whatever.txt 
PHPUnit 4.0.12 by Sebastian Bergmann.

Configuration read from /Users/taejun/Code/devtools/phpunit.xml

..taejun@mbp-002672 devtools (dev) $ 
```

and I see here that the terminal prompt is on the same line as the cat output.